src/checkpoint/checkpoint.h

```diff
@@ -65,43 +65,44 @@ struct __wt_ckpt_thread {
     uint64_t usecs;             /* thread timer */
 };
 
+/*
+ * WT_CKPT_TIME_STATS --
+ *     Time-related statistics.
+ */
+struct __wt_ckpt_time_stats {
+    struct timespec timer_end;
+    struct timespec timer_start;
+    uint64_t max;
+    uint64_t min;
+    uint64_t recent;
+    uint64_t total;
+};
+
 /*
  * WT_CKPT_CONNECTION --
  *     Checkpoint information.
  */
 struct __wt_ckpt_connection {
-    WT_CKPT_THREAD server; /* Checkpoint thread.*/
-
-    wt_shared uint64_t most_recent; /* Clock value of most recent checkpoint */
 
     WT_CKPT_HANDLE_STATS handle_stats;
 
-    uint64_t scrub_max; /* Checkpoint scrub time min/max */
-    uint64_t scrub_min;
-    uint64_t scrub_recent; /* Checkpoint scrub time recent/total */
-    uint64_t scrub_total;
-
-    uint64_t prep_max; /* Checkpoint prepare time min/max */
-    uint64_t prep_min;
-    uint64_t prep_recent; /* Checkpoint prepare time recent/total */
-    uint64_t prep_total;
-    uint64_t time_max; /* Checkpoint time min/max */
-    uint64_t time_min;
-    uint64_t time_recent; /* Checkpoint time recent/total */
-    uint64_t time_total;
-
-    /* Checkpoint stats and verbosity timers */
-    struct timespec prep_end;
-    struct timespec prep_start;
-    struct timespec timer_start;
-    struct timespec timer_scrub_end;
+    /* Checkpoint thread. */
+    WT_CKPT_THREAD server;
+
+    /* Time-related stats. */
+    WT_CKPT_TIME_STATS prepare;
+    WT_CKPT_TIME_STATS scrub;
+    WT_CKPT_TIME_STATS time;
```

Comment:
I don't really like the general term "time". This one is related to general timing stats, what would be better?

Reply:
Me too, it seems too broad. Can you rename it to something more specific to the use case? What does this time represent?

Reply:
It's general stats about the checkpoint: the min/max time it has ever taken, the time of the current one. While the other fields are for dedicated stages of the checkpoint.

Reply:
Perhaps, `total_time` then?

Reply:
or just `total`?

Reply:
If we had `total` it would be weird as we have a field already named `total` in the struct, we would have `ckpt.total.total`.
I will wait for the others to chime in.

Reply:
What about:

1. `overall`
2. `summary`
3. `aggregate`

Reply:
What about `ckpt_api` or `ckpt_op` as the time for the operation? Of your list, I slightly lean toward `aggregate` although it isn't strictly aggregating the other times.